sample_apps/simple_app/run_example.sh

```diff
@@ -80,6 +79,7 @@ function usage() {
 # ###########################################################################
 Steps=("show_env"
         "do_cleanup"
+        "rm_non_git_files"
```

Comment:
We should consider an option here:

Currently, with this fix, we will delete all non-Git files each time you run this script.

John wanted some flexibility to retain outputs of previously run steps, so this script's execution can be optimized.

That can be 'controlled' by location of this new step `rm_non_git_files` in this area.

I can move it to the top. And do the following:

- If you run `run_example.sh` by itself, it will start from the top, so `rm_non_git_files` will be invoked.
- If you run `run_example.sh setup`, that's an indication that we may want to retain existing artifacts. So execution will begin from 2nd step, i.e. `show_env`. 

This way, one gets a bit more flexibility of how this script can be used.

In CI, we are running `./run_example.sh` without any arguments, so you will get the rm-non-Git-objects behaviour.

Let me know what you prefer.

Reply:
I think this makes sense. I'll leave it to John.

Reply:
@jlmucb - To expedite decision-making, I decided to go with this new proposed change.

`run_example.sh` w/o any args will always invoke the newly added `rm_non_git_files()` function, so we will now exercise this script from  a clean-cloned repo build-env.

I also extended `build.yml` to exercise this combination of `setup`, followed by one or more `run_test` commands.

Once this works in CI, we can have more confidence that this interface will stay stable and useful / usable by customers.